### PR TITLE
Added check_dirty.py to check if a repo is dirty

### PR DESCRIPTION
**Pull request by almahans**
Added a script to check if a given repository is dirty, to make sure that the correct version is run and no changes have been made to the code locally.

### CODE REVIEW

File: check_dirty.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"""
```
Comment (alneberg) [inline check_dirty.py:1]: I consider it best practice to include a "shebang" here: https://realpython.com/python-shebang/#whats-a-shebang-and-when-should-you-use-it

Comment (alneberg) [reply check_dirty.py:1]: It has to be the first row of the file I think.

File: check_dirty.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	repo = Repo(args.repo_path)
+	if repo.is_dirty():
+		print("Repo is dirty.", file=sys.stderr)
+
+if __name__ == "__main__":
+	parser = argparse.ArgumentParser(description='check if dirty')
+	parser.add_argument('repo_path', metavar='P', type=str, help='path to repository to check')
+	args = parser.parse_args()
+
+	check_if_dirty(args)
```
Comment (alneberg) [inline check_dirty.py:18]: I think I prefer to have:
```
check_if_dirty(args.repo_path)
```
and then
```
def check_if_dirty(repo_path):
    repo = Repo(repo_path)
```

File: check_dirty.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+"""
+Checks if a repo is dirty.
+"""
+import argparse
+import sys
+from git import Repo
+
+def check_if_dirty(repository):
+	repo = Repo(args.repo_path)
+	if repo.is_dirty():
```
Comment (alneberg) [inline check_dirty.py:10]: Also, can you switch to 4 spaces indentation? That's what we seem to use most eventhough I don't think it's a defined standard anywhere.

File: check_dirty.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+Checks if a repo is dirty.
+"""
+import argparse
+import sys
+from git import Repo
+
+def check_if_dirty(repo_path):
+    repo = Repo(repo_path)
+    if repo.is_dirty():
+	    print("Repo is dirty.", file=sys.stderr)
```
Comment (alneberg) [inline check_dirty.py:12]:
```suggestion
    if repo.is_dirty():
        print("Repo is dirty.", file=sys.stderr)
```

File: check_dirty.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    repo = Repo(repo_path)
+    if repo.is_dirty():
+        print("Repo is dirty.", file=sys.stderr)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='check if dirty')
+    parser.add_argument('repo_path', metavar='P', type=str, help='path to repository to check')
+    args = parser.parse_args()
+
+    check_if_dirty(args.repo_path)
```
Comment (almahans) [inline check_dirty.py:19]:
```suggestion
    check_if_dirty(args.repo_path)

```

**Review by alneberg — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**